### DISCUSSION TITLE
How to use nest query in nest store?

### DISCUSSION THREAD

**Discussion by hh54188** [salesforce/akita#565] [Q&A]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[x] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

I try to nest a store in another store, so I need to query the nested store as well. Here is my demo code :

```javascript

export class Students extends EntityStore<StudentState> {
    constructor() {
        super();
    }
}

export class StudentsQuery extends  QueryEntity<StudentState> {
    constructor(protected store) {
        super(store);
    }
}

export class ClassRoom extends Store<ClassRoomState> {
    constructor() {
        super({
            name: '',
            students: new Students()
        });
    }
}

export class ClassRoomQuery extends Query<ClassRoom> {
    constructor(protected store) {
        super(store);
    }
    getStudents() {
        return this.select((state: MyFoldersState) => state.students)
            .subscribe(students$ => new StudentsQuery(students$).selectAll()
                .subscribe(result => console.log(result)));
    }
}

```

The `Students` is the nest store, it's used as a field in `ClassRoom` store. I need to query its data from `ClassRoom` store, so I write the `getStudents` method in `ClassRoomQuery`.

I'm not sure the way I write this method is correct. When I call it, it gave me an error:

```
core.js:5973 ERROR TypeError: Cannot add property 0, object is not extensible
    at Array.push (<anonymous>)
    at BehaviorSubject._subscribe (Subject.js:97)
    at BehaviorSubject._subscribe (BehaviorSubject.js:12)
    at BehaviorSubject._trySubscribe (Observable.js:42)
    at BehaviorSubject._trySubscribe (Subject.js:81)
    at BehaviorSubject.subscribe (Observable.js:28)
    at Observable._subscribe (Observable.js:76)
    at Observable.subscribe (Observable.js:27)
    at MapOperator.call (map.js:16)
    at Observable.subscribe (Observable.js:23)
```

How can I do it right?

## Expected behavior

Query the nest store's data successfully.

## Minimal reproduction of the problem with instructions

See the demo code above.

## What is the motivation / use case for changing the behavior?

Nest store is only for entity collection. There are some other properties not suit there .

## Environment

<pre><code>
Angular version: 11.0.2

Browser:
- [x] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 12.11.0
- Platform:  Windows

**Comment by hakimio** [comment]
I don't think you understand how RxJS and Akita works.
Here is my interpretation of what you are trying to achieve:
```ts
export interface ClassRoom {
    id: number;
    name: string;
    studentIds: number[];
}

export interface ClassRoomState extends EntityState<ClassRoom>, ActiveState {}

@Injectable({
    providedIn: 'root'
})
export class ClassRoomQuery extends QueryEntity<ClassRoomState> {
    
    activeClassRoom$ = this.selectActive();
    
    students$ = activeClassRoom$
        .pipe(
            switchMap(
                classRoom => this.studentQuery.selectMany(classRoom.studentIds)
            )
        );
        
    constructor(
        protected readonly store: ClassRoomStore,
        private readonly studentQuery: StudentQuery
    ) {
        super(store);
    }

}
```

You need to learn how RxJS/Observables work before you can use Akita effectively.
Also, reading [Working with Normalized Data in Akita and Angular](https://netbasal.com/working-with-normalized-data-in-akita-e626d4c67ca4) article might be useful for you as well.
